Resolving Keccak-256: check if arguments are identifiers early. (#11899)
ethereum/solidity: Changelog.md

 * SMTChecker: Add constraints to better correlate ``address(this).balance`` and ``msg.value``.
 * SMTChecker: Support the ``value`` option for external function calls.
 * Commandline Interface: Disallowed the ``--experimental-via-ir`` option to be used with Standard Json, Assembler and Linker modes.

>  * Yul Optimizer: fix a crash in LoadResolver.

Reviewer: ```suggestion
 * Yul Optimizer: Fix a crash in LoadResolver.
```
Resp. could elaborate a bit more...

>  * Yul Optimizer: Fix a crash in LoadResolver, when ``keccak256`` had particular arguments.

Reviewer 2: ```suggestion
 * Yul Optimizer: Fix a crash in LoadResolver when ``keccak256`` has non-identifier arguments.
```

Author: But not every non-identifier arguments? Maybe I'll change to "has particular non-identifier arguments"?

Author: Changed it. Writing the changelog for this was more work than the fix :)

>  * Yul Optimizer: Fix a crash in LoadResolver, when ``keccak256`` has particular non-identifier arguments.

Reviewer 3: It's a bugfix.

Reviewer: Shame on me.

Author: Oops. Fixed in https://github.com/ethereum/solidity/pull/11901
